### PR TITLE
Communications should be in a container object

### PR DESCRIPTION
**Pull request by jdfreder**
woot

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,59 @@
...
+
 The core functionality of the notebook is to send messages from a frontend to a backend, and from a backend to a frontend ([or many frontends](https://github.com/nteract/jupyter-sidecar)). In the case of the Jupyter/IPython notebook, it communicates over websockets (which in turn reach out to ØMQ on the backend).
 
 What if you want to serve the same HTML and Javascript for the notebook application itself while being able to work in a native ØMQ environment? What if websockets are fairly restricted in your working \*ahem\* *corporate* environment and you need to send data via `POST` and receive streaming updates using server-sent events?
 
 We'd need a nice clean way to abstract the transport layer. Since [Jupyter is messages all the way down](http://jupyter-client.readthedocs.org/en/latest/messaging.html), one way is to hook up a series of event emitters all with the same interface. That's [definitely do-able](https://github.com/nteract/jupyter-transport-wrapper). Instead, let's rely on Observables: asynchronous data streams, [*from the future*](https://zenparsing.github.io/es-observable/). Observables are the multi-valued promise we've all been waiting for:
+To spawn a kernel
```
Comment (rgbkrk) [inline README.md]: I'm assuming you didn't mean to include the line above here.

File: README.md [hunk 2]
```diff
@@ -1,30 +1,59 @@
...
-   </tr>
-   <tr>
-      <td>Push/Asynchronous/Reactive</td>
-      <td>Promise</td>
-      <td>Observable</td>
-   </tr>
-</table>
-
-Even better is to rely on RxJS's implementation, since we get a nice way to handle these streams:
+*[For more information see the Jupyter client docks.](http://jupyter-client.readthedocs.org/en/latest/messaging.html)*
```
Comment (rgbkrk) [inline README.md]: s/docks/docs/

File: README.md [hunk 3]
```diff
@@ -1,30 +1,59 @@
...
+At the heart of enchannel is the `kernel` [subject](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/gettingstarted/subjects.md).  To implement the enchannel spec and be an *enchannel-backend*, a *backend* must export a `kernel` subject.  This is the only top-level API required of *enchannel-backends*.  The `kernel` subject manages the life cycles of kernels.  To spawn a kernel, submit the name of the kernel spec to the subject:
+
+```js
+kernel.next('python3');
+```
+
+Spawning another kernel will close the last.  To explicitly close a kernel and/or when you're done with the `kernel` subject, complete the `kernel` subject:
+
+```js
+kernel.onCompleted();
```
Comment (rgbkrk) [inline README.md]:
```
kernel.complete() // RxJS 5 / ES 7 Observable
```

File: README.md [hunk 4]
```diff
@@ -1,30 +1,59 @@
...
+```
```
Comment (rgbkrk) [inline README.md]: Q: What would `kernel.error` be?

**Comment by rgbkrk** [comment]
I like this approach to lifecycles of kernels. Our 80% use case tends to be launch one kernel and complete it. The reason someone swaps kernels is to switch their notebook on initial launch to their real intent and in some cases to test their code on different pythons.

Now... I'm not sure if all backends have to have a strict requirement of kernel name. I imagine a very near future where we'll be able to specify a github repo (binder!) or a docker container (muahahaha carina). The way someone configures that backend would be up to them and the docs for that backend.

Especially important here would be when a backend has to rely on auth via cookie, token, or PKI. What the enchannel spec is driving is consistency on the kernel messaging. I'm slightly thinking this is a separate spec for the spawning or at least a utility helper to close down channels?

**Comment by jdfreder** [comment]
I think there are two things missing here:
1. The ability to create more than one `kernel` subject.
2. The ability to list kernel spec names (or docker image names, etc)

File: README.md (hunk 1 above)
Comment (jdfreder) [inline README.md]: :heavy_check_mark:

File: README.md (hunk 2 above)
Comment (jdfreder) [inline README.md]: :heavy_check_mark:

File: README.md (hunk 3 above)
Comment (jdfreder) [inline README.md]: deleted

File: README.md (hunk 4 above)
Comment (jdfreder) [inline README.md]: deleted

**Comment by jdfreder** [comment]
Okay this is good to go

**Comment by rgbkrk** [comment]
Cool, good spec updates.

**Pull request merged**